src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -181,134 +184,198 @@ private boolean isSeqColL1Present(SeqColLevelOneEntity levelOneEntity) {
+        List<String> seqColBAttributesList = new ArrayList<>(seqColBAttributeSet); // For better data manipulation
+
         // "arrays" attribute
-        try {
-            Field[] seqColAFields = seqColAEntity.getClass().getDeclaredFields();
-            Field[] seqColBFields = seqColBEntity.getClass().getDeclaredFields();
-            List<String> seqColAFieldNames = getFieldsNames(seqColAFields);
-            List<String> seqColBFieldNames = getFieldsNames(seqColBFields);
-            List<String> seqColAUniqueFields = getUniqueFields(seqColAFieldNames, seqColBFieldNames);
-            List<String> seqColBUniqueFields = getUniqueFields(seqColBFieldNames, seqColAFieldNames);
-            List<String> seqColCommonFieldNames = getCommonFieldsDistinct(seqColAFieldNames, seqColBFieldNames);
-            comparisonResult.putIntoArrays("a-only", seqColAUniqueFields);
-            comparisonResult.putIntoArrays("b-only", seqColBUniqueFields);
-            comparisonResult.putIntoArrays("a-and-b", seqColCommonFieldNames);
-
-        } catch (NullPointerException e) {
-            // TODO LOGGER MESSAGE
-            System.out.println("Either seqColA or seqColB is not initialized");
-            throw new RuntimeException(e.getMessage());
-        }
-        // "elements"
-        List<String> seqColALengths = seqColAEntity.getLengths();
-        List<String> seqColBLengths = seqColBEntity.getLengths();
-        List<String> seqColANames = seqColAEntity.getNames();
-        List<String> seqColBNames = seqColBEntity.getNames();
-        List<String> seqColASequences = seqColAEntity.getSequences();
-        List<String> seqColBSequences = seqColBEntity.getSequences();
+        List<String> seqColAUniqueAttributes = getUniqueElements(seqColAAttributesList, seqColBAttributesList);
+        List<String> seqColBUniqueAttributes = getUniqueElements(seqColBAttributesList, seqColAAttributesList);
+        List<String> seqColCommonAttributes = getCommonElementsDistinct(seqColAAttributesList, seqColBAttributesList);
+        comparisonResult.putIntoArrays("a-only", seqColAUniqueAttributes);
+        comparisonResult.putIntoArrays("b-only", seqColBUniqueAttributes);
+        comparisonResult.putIntoArrays("a-and-b", seqColCommonAttributes);
 
         // "elements" attribute | "total"
-        Integer seqColATotal = seqColAEntity.getSequences().size();
-        Integer seqColBTotal = seqColBEntity.getSequences().size();
+        String seqColARandomAttribute = seqColAAttributesList.get(0); // could be any attribute, normally they all have the same size
+        String seqColBRandomAttribute = seqColBAttributesList.get(0); // could be any attribute, normally they all have the same size
+        Integer seqColATotal = seqColAEntityMap.get(seqColARandomAttribute).size();
+        Integer seqColBTotal = seqColBEntityMap.get(seqColBRandomAttribute).size();
```

Comment:
I think I would prefer sticking to the using either `sequences`, `name` or `length`. It is in theory possible that other attributes have different lengths.

Reply:
Yeah that makes sense.
I think we can also add some other tests in the future regarding the seqCol attributes provided in the POST request, because the end user might not always provide correct seqCol data...but it needs some sort of rules (eg: in the specification) on **required** and **non required** attributes..

Reply:
However, the reason why I did choose a random attribute is because I assumed that we don't know the attributes that the seqCol actually contains, and we will handle them generically (regardless of the seqCol attributes).
But I assume that `sequences`, `names` or `lengths` are required attributes of every seqCol object, so they must be always present, so it make sense to call them explicitly in this case. I don't know if I'm correct.

Reply:
That's a fair point. 
There is no guarantee that a sequence collection provided in POST would be containing mandatory fields since we do not validate their presence so you should be checking for `lengths` (which should always be present in a valid SeqCol) and if absent you should raise an exception that sends back a 400 error. Maybe 411 although that might not be right.